x-pack/docs/en/rest-api/security/update-api-key.asciidoc

```diff
@@ -1,5 +1,277 @@
 [role="xpack"]
 [[security-api-update-api-key]]
-=== Update API key information API
+=== Update API key API
 
-coming::[8.4.0]
+++++
+<titleabbrev>Update API key</titleabbrev>
+++++
+
+[[security-api-update-api-key-request]]
+==== {api-request-title}
+
+`PUT /_security/api_key/<id>`
+
+[[security-api-update-api-key-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_own_api_key` cluster privilege.
+Users can only update API keys that they created or that were granted to them.
+To update another user's API key, use the <<run-as-privilege,`run_as` feature>>
+to submit a request on behalf of another user.
+
+IMPORTANT: It's not possible to use an API key as the authentication credential for this API.
+To update an API key, the owner user's credentials are required.
+
+[[security-api-update-api-key-desc]]
+==== {api-description-title}
+
+Use this API to update API keys created by the <<security-api-create-api-key,create API Key>> or <<security-api-grant-api-key,grant API Key>> APIs.
+It's not possible to update expired API keys, or API keys that have been invalidated by <<security-api-invalidate-api-key,invalidate API Key>>.
+
+This API supports updates to an API key's access scope, i.e., its permissions, and its metadata.
+The access scope of an API key is derived from the <<security-api-update-api-key-api-key-role-descriptors,`role_descriptors`>> you specify in the request, and a snapshot of the owner user's permissions at the time of the request.
+The snapshot of the owner's permissions is updated automatically on every call.
+
+[IMPORTANT]
+====
+A call to this API might change the API key's access scope, even if you don't specify new
+<<security-api-update-api-key-api-key-role-descriptors,`role_descriptors`>> in the request.
+This behavior can occur if the owner user's permissions have changed since the last modification of the API key.
+====
+
+[[security-api-update-api-key-path-params]]
+==== {api-path-parms-title}
+
+`id`::
+(Required, string) The ID of the API key to update.
+
+[[security-api-update-api-key-request-body]]
+==== {api-request-body-title}
+(Optional)
+
+You can specify the following optional parameters in the request body.
+
+[[security-api-update-api-key-api-key-role-descriptors]]
+`role_descriptors`::
+(Optional, array) An array of role descriptors to assign to this API key.
```

Comment:
This is technically incorrect because this parameter does not accept array, but must be an object. Create API Key page has the same issue.

Reply:
Indeed, forgot to comment on this. Thanks for flagging! I can change this to `(Optional, object) The role descriptors to assigned to this API key` and likewise tweak the create and grant docs.